disk/disk.go

```diff
@@ -34,7 +39,7 @@ func WriteAt(path string, p []byte, off int64) (int, error) {
 		data = append(make([]byte, padding), p...)
 	}
 	// index that ends writing
-	end := st + int64(len(data))
+	end := st + int64(len(data)) - 1
```

Comment:
@yutongp What is the reason to change this? it should be the index that exactly after the last byte written in my mind.(The comment is confused, sry)

If we go this way, we need to change some related code, e.g., `if endOff > 0 {` to `if endOff >= 0 {`. But let us discuss about the possible bug cases caused by this `end` and the best solution to solve it first.